src/mento.ts

```diff
@@ -0,0 +1,260 @@
+import { BigNumber, ContractTransaction, Signer } from 'ethers'
+import {
+  IBroker,
+  IBroker__factory,
+  IExchangeProvider,
+  IExchangeProvider__factory,
+} from '@mento-protocol/mento-core-ts'
+import { Provider, TransactionResponse } from '@ethersproject/providers'
+import {
+  getBrokerAddressFromRegistry,
+  getSymbolFromTokenAddress,
+} from './utils'
+
+import { strict as assert } from 'assert'
+
+interface Exchange {
+  providerAddr: string
+  id: string
+  assets: string[]
+}
+
+interface Asset {
+  address: string
+  symbol: string
+}
+
+export class Mento {
+  private readonly provider: Provider
+  private readonly signer?: Signer
+  private readonly broker: IBroker
+  private exchanges: Exchange[]
+
+  /**
+   * This constructor is private, use the static create method to create a new Mento instance
+   * @param brokerAddress the address of the broker contract
+   * @param provider an ethers provider
+   * @param signer an optional ethers signer to execute swaps (must be connected to a provider)
+   */
+  private constructor(
+    brokerAddress: string,
+    provider: Provider,
+    signer?: Signer
+  ) {
+    this.broker = IBroker__factory.connect(brokerAddress, provider)
+    this.provider = provider
+    this.signer = signer
+
+    this.exchanges = new Array<Exchange>()
+  }
+
+  /**
+   * Create a new Mento object instance
+   * @param ethersProvider an ethers provider
+   * @param ethersSigner an optional ethers signer to execute swaps (must be connected to a provider)
+   * @returns a new Mento object instance
+   */
+  static async create(ethersProvider: Provider, ethersSigner?: Signer) {
+    return new Mento(
+      await getBrokerAddressFromRegistry(ethersProvider),
+      ethersProvider,
+      ethersSigner
+    )
+  }
+
+  /**
+   * Returns a list of all the pairs that can be traded on Mento
+   * @returns The list of tradeable pairs in the form of [{address, symbol}]
+   */
+  async getTradeablePairs(): Promise<[Asset, Asset][]> {
+    const exchanges = await this.getExchanges()
+    let pairs: [Asset, Asset][] = []
+
+    for (let exchange of exchanges) {
+      const asset0 = exchange.assets[0]
+      const asset1 = exchange.assets[1]
+      let symbols = await Promise.all([
```

Comment:
symbols could be `const`, recommend setting up lint and/or ts rules to enforce that. 
A reasonable ruleset example here: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/.eslintrc

Reply:
Agree, would just fix it here right now and create an issue for the linting rules